ambari-server/src/main/java/org/apache/ambari/server/utils/SecretReference.java

```diff
@@ -150,6 +166,49 @@ public static void replacePasswordsWithReferences(Map<PropertyInfo.PropertyType,
     }
   }
 
+  /**
+   * Returns all password properties defined in the stacks specified by the given stack id's. Keys in the map are
+   * file names (e.g hadoop-env.xml) and values are property names.
+   * @param stackIds the stack ids to specify which stacks to look for
+   * @return A set multimap of password type properties.
+   * @throws IllegalArgumentException when a non-existing stack is specified
+   */
+  public static SetMultimap<String, String> getAllPasswordProperties(Collection<StackId> stackIds) throws IllegalArgumentException {
+    AmbariMetaInfo metaInfo = AmbariServer.getController().getAmbariMetaInfo();
+    Collection<StackInfo> stacks = stackIds.stream().map( stackId -> {
+      try {
+        return metaInfo.getStack(stackId);
+      }
+      catch (StackAccessException ex) {
+        throw new IllegalArgumentException(ex);
+      }
+    }).collect(toList());
+    return getAllPasswordPropertiesInternal(stacks);
+  }
+
+  /**
+   * Returns all password properties defined in all stacks. Keys in the map are
+   * file names (e.g hadoop-env.xml) and values are property names.
+   * @return A set multimap of password type properties.
+   */
+  public static SetMultimap<String, String> getAllPasswordProperties() {
+    AmbariMetaInfo metaInfo = AmbariServer.getController().getAmbariMetaInfo();
+    return getAllPasswordPropertiesInternal(metaInfo.getStacks());
+  }
+
+  private static SetMultimap<String, String> getAllPasswordPropertiesInternal(Collection<StackInfo> stacks) {
```

Comment:
Making this non-private would allow testing without the need for PowerMock.

Reply:
Powermock is needed because of the static call to AmbariServer.getController(), not because this method is private.

Reply:
I meant that this method could be tested instead of the one that calls `AmbariServer.getController()`.  Most of the logic is in this one.

Reply:
Ok.